scripts/process_schemas.py

```diff
@@ -201,44 +207,60 @@ def write(self, outputFile):
             superclass = "SearchRequest"
         elif self.isSearchResponse():
             superclass = "SearchResponse"
-        string = "\n\nclass {0}({1}):".format(self.schema.name, superclass)
+        self._writeNewline(outputFile, 2)
+        string = "class {0}({1}):".format(self.schema.name, superclass)
         print(string, file=outputFile)
         doc = self.schema.doc
         if doc is None:
             doc = "No documentation"
-        string = '    """\n{0}\n    """'.format(doc)
-        print(string, file=outputFile)
+        self._writeWithIndent('"""', outputFile)
+        self._writeWrappedWithIndent(doc, outputFile)
+        self._writeWithIndent('"""', outputFile)
         if isinstance(self.schema, avro.schema.RecordSchema):
-            string = '    _schemaSource = """\n{0}"""'.format(
+            string = '_schemaSource = """\n{0}"""'.format(
                 self.formatSchema())
-            print(string, file=outputFile)
-            string = '    schema = avro.schema.parse(_schemaSource)'
-            print(string, file=outputFile)
-            string = '    requiredFields = {0}'.format(
-                self.formatRequiredFields())
-            print(string, file=outputFile)
+            self._writeWithIndent(string, outputFile)
+            string = 'schema = avro.schema.parse(_schemaSource)'
+            self._writeWithIndent(string, outputFile)
+            self.writeRequiredFields(outputFile)
             if self.isSearchResponse():
-                string = '    _valueListName = "{0}"'.format(
+                string = '_valueListName = "{0}"'.format(
                     self.getValueListName())
-                print(string, file=outputFile)
-            print(file=outputFile)
+                self._writeWithIndent(string, outputFile)
+            self._writeNewline(outputFile)
             self.writeEmbeddedTypesClassMethods(outputFile)
             self.writeConstructor(outputFile)
         elif isinstance(self.schema, avro.schema.EnumSchema):
             # TODO make a proper Python enum here using the Python 3.4 enum?
             for symbol in self.schema.symbols:
-                string = '    {0} = "{0}"'.format(symbol, symbol)
-                print(string, file=outputFile)
+                string = '{0} = "{0}"'.format(symbol, symbol)
+                self._writeWithIndent(string, outputFile)
+
+    def _writeWithIndent(self, string_, outputFile, indentLevel=1):
```

Comment:
Very nice, excellent abstraction. We seem to just use `outputFile` as a parameter all the time: do we ever write to more than one file? Perhaps we should make a an instance variable `self._outputFile`, which might tidy things up even more?